portal-bridge/src/api/execution.rs

```diff
@@ -51,7 +51,7 @@ pub struct ExecutionApi {
 }
 
 impl ExecutionApi {
-    pub async fn new(primary: Url, fallback: Url) -> Result<Self, surf::Error> {
+    pub async fn new(primary: Url, fallback: Url) -> Result<Self, reqwest_middleware::Error> {
         // Only check that provider is connected & available if not using a test provider.
         debug!(
             "Starting ExecutionApi with primary provider: {primary} and fallback provider: {fallback}",
```

Comment:
Comment on line 276 should be removed, and probably some testing to make sure that `reqwest` can handle the async load, though it's a very popular library so I expect it will